File: Javascript/Daily Coding Problem#9.js
const findLargestSum = (arr) => {
    let cache = new Array(arr.length);
    let size = arr.length;
    cache[size - 1] = arr[size - 1];
    for (let i = size - 2; i >= 0; i--) {
        if (i == size - 2) {
            cache[i] = arr[i] > arr[i + 1] ? arr[i] : arr[i + 1];
        } else {
            cache[i] = Math.max((arr[i] + cache[i + 2]), cache[i + 1]);
        }
    }
    return cache[0];
}

console.log(findLargestSum([3, -4, -10, -5, -10, 6]));

File: Python/Daily Coding Problem#9.py
def findLargestSum(arr):
    cache = [0] * (len(arr))
    for i in reversed(range(len(arr))):
        if i == len(arr) - 1:
            cache[i] = arr[i]
        elif i == len(arr) - 2:
            cache[i] = arr[i] if arr[i] > arr[i+1] else arr[i+1]
        else:
            cache[i] = max((arr[i] + cache[i+2]), cache[i+1])
    return cache[0]


size = int(input("Enter size of an array"))
print("Enter array elements")
nums = []
for i in range(0, size):
    nums.append(int(input()))

print(findLargestSum(nums))
